attunement.lic

```diff
@@ -9,21 +9,41 @@ class Attunement
   include DRCA
   include DRCT
 
+  LUNAR_PERCEIVE_COMMANDS = ['mana', 'moons', 'planets']
+
   def initialize
     settings = get_settings
     @stationary_skills_only = settings.crossing_training_stationary_skills_only
     @hometown = settings.hometown
     @attunement_rooms = settings.attunement_rooms
-    @start_timer = Time.now
+    @exp_timer = settings.exp_timers['Attunement'] || 90
+    @exp_max_threshold = settings.crossing_training_max_threshold
+    @lunar_perceive_index = -1
     train_attunement(settings)
   end
 
+  def is_lunar_mage?
+    DRStats.moon_mage? || DRStats.trader?
+  end
+
+  def get_next_lunar_perceive_index
+    @lunar_perceive_index = @lunar_perceive_index + 1
```

Comment:
The modulus operator (`%`) could simplify this logic

```
@commands = ['a', 'b', 'c']
@index = 0

def increment_index()
  @index = (@index + 1) % @commands.length
end

puts @index
increment_index()
puts @index
increment_index()
puts @index
increment_index()
puts @index
increment_index()
puts @index
increment_index()
puts @index
increment_index()
puts @index
increment_index()
```

Gives:
```
> ruby main.rb
0
1
2
0
1
2
0
```

Reply:
Sweet, thanks!